python/perf-kernels/streamk/tune_streamk.py

```diff
@@ -0,0 +1,770 @@
+# fp8
+import argparse
+import sys
+import yaml
+import os
+import glob
+import subprocess
+
+import torch
+import triton
+import triton.language as tl
+
+from streamk_kernel import streamk_gemm
+
+from datetime import datetime
+import multiprocessing
+import pandas as pd
+
+
+def get_full_tuning_space():
+    configs = []
+
+    block_mn_range = [16, 32, 64, 128, 256]
+    block_k_range = [16, 32, 64, 128, 256]
+    num_warps_range = [1, 2, 4, 8]
+    group_m_range = [1, 4, 8, 16, 32]
+    # For now we see better perf with num_stages=0 for all gemm configs we care
+    # But keep this explicit so that we do not forget we may need to set it to
+    # other values in the future
+    num_stage_range = [0]
+    waves_per_eu_range = [0]
+    matrix_instr_nonkdim_range = [16, 32]
+    kpack_range = [1, 2]
+
+    for block_m in block_mn_range:
```

Comment:
alternatives to avoid 9-level indent:

suggested by chatgpt-4o:

```python
import itertools

def get_full_tuning_space():
    block_mn_range = [16, 32, 64, 128, 256]
    block_k_range = [16, 32, 64, 128, 256]
    num_warps_range = [1, 2, 4, 8]
    group_m_range = [1, 4, 8, 16, 32]
    num_stage_range = [0]
    waves_per_eu_range = [0]
    matrix_instr_nonkdim_range = [16, 32]
    kpack_range = [1, 2]

    ranges = [block_mn_range, block_mn_range, block_k_range, num_warps_range, 
              group_m_range, num_stage_range, waves_per_eu_range, 
              matrix_instr_nonkdim_range, kpack_range]

    keys = ['BLOCK_SIZE_M', 'BLOCK_SIZE_N', 'BLOCK_SIZE_K', 'num_warps', 
            'GROUP_SIZE_M', 'num_stages', 'waves_per_eu', 
            'matrix_instr_nonkdim', 'kpack']

    configs = [dict(zip(keys, values)) for values in itertools.product(*ranges)]

    return configs
```

or meta.ai:

```python
def get_full_tuning_space():
    block_mn_range = [16, 32, 64, 128, 256]
    block_k_range = [16, 32, 64, 128, 256]
    num_warps_range = [1, 2, 4, 8]
    group_m_range = [1, 4, 8, 16, 32]
    num_stage_range = [0]
    waves_per_eu_range = [0]
    matrix_instr_nonkdim_range = [16, 32]
    kpack_range = [1, 2]

    configs = [
        {
            'BLOCK_SIZE_M': block_m,
            'BLOCK_SIZE_N': block_n,
            'BLOCK_SIZE_K': block_k,
            'GROUP_SIZE_M': group_m,
            'num_warps': num_warps,
            'num_stages': num_stages,
            'waves_per_eu': waves_per_eu,
            'matrix_instr_nonkdim': matrix_instr_nonkdim,
            'kpack': kpack
        }
        for block_m in block_mn_range
        for block_n in block_mn_range
        for block_k in block_k_range
        for num_warps in num_warps_range
        for group_m in group_m_range
        for num_stages in num_stage_range
        for waves_per_eu in waves_per_eu_range
        for matrix_instr_nonkdim in matrix_instr_nonkdim_range
        for kpack in kpack_range
    ]

    return configs
```